Source/components/SymbolInstance.js

```diff
@@ -0,0 +1,114 @@
+import { DefinedPropertiesKey } from '../WrappedObject'
+import { Layer } from './Layer'
+import { Rectangle } from '../Rectangle'
+import { Types } from '../enums'
+import { Factory } from '../Factory'
+import { wrapObject } from '../wrapNativeObject'
+import { toArray } from '../utils'
+import { Override } from './Override'
+
+/**
+ * A Sketch symbol instance.
+ */
+export class SymbolInstance extends Layer {
+  /**
+   * Make a new symbol instance.
+   */
+  constructor(master = {}) {
+    if (!master.sketchObject) {
+      // eslint-disable-next-line no-param-reassign
+      master.sketchObject = Factory.createNative(SymbolInstance)
+        .alloc()
+        .initWithFrame(new Rectangle(0, 0, 100, 100).asCGRect())
+    }
+    super(master)
+  }
+
+  // Replaces the instance with a group that contains a copy of the Symbol this instance refers to. Returns null if the master contains no layers instead of inserting an empty group
+  detach() {
+    const group = this._object.detachByReplacingWithGroup()
+
+    if (group) {
+      return wrapObject(group)
+    }
+
+    return null
+  }
+
+  // Returns a new SymbolInstance linked to this artboard, ready for inserting in the document
+  createNewInstance() {
+    return SymbolInstance.fromNative(this._object.newSymbolInstance())
+  }
+
+  // Returns all instances of the artboard in the document, on all pages
+  getAllInstances() {
+    return toArray(this._object.allInstances()).map(SymbolInstance.fromNative)
+  }
+
+  setOverrideValue(override, value) {
+    this._object.setValue_forOverridePoint(
+      value,
+      wrapObject(override).sketchObject.overridePoint()
+    )
+  }
+}
+
+SymbolInstance.type = Types.SymbolInstance
+SymbolInstance[DefinedPropertiesKey] = { ...Layer[DefinedPropertiesKey] }
+Factory.registerClass(SymbolInstance, MSSymbolInstance)
+
+SymbolInstance.define('symbolId', {
+  get() {
+    return String(this._object.symbolID())
+  },
+  set(id) {
+    // we need to find the symbol master and change the master,
+    // it's not enough to just call `this._object.setSymbolID`
+    const master = this._object
+      .parentPage()
+      .documentData()
+      .symbolWithID(id)
+
+    this.parent = master
```

Comment:
Not sure I understand this - is this to force a refresh? I think the parent for a symbol instance would always be the page (or artboard, group, etc) that contains the symbol instance so I don't think we'd want to set it to be the master.

You might be better off calling `changeInstanceToSymbol:` for this setter instead.

Reply:
Just read a bit further down - would we be better off having this a readonly property so that they go through `setMaster`?

Reply:
It's a typo (haven't tested everything yet), it should be `this.master = master` which would calls the `set(master)` defined underneath.

I don't really like having read-only properties, it's not possible to know that it's read-only in javascript.
So I'm usually creating a method (`instance.getSymbolId()`) so that you know that you can't set it or creating a proper property. In this case, it's possible to have a property so I decided to go with it.

But yes, behind the scene it's just a helper for the master setter.